core_v2/sources/readme.md

```diff
@@ -0,0 +1,100 @@
+# Aptos Naming Service
+
+The Aptos Naming Service is a decentralized naming service for the Aptos blockchain.
+At its core, the naming service is an on-chain contract that allows users to register domain and subdomain names, and
+point them to an account. This allows users to have a human-readable name for their account, and to use it in place of
+the long and un-ergonomic hex addresses that exist today.
+
+For example, instead of sending a transaction to `0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef`,
+you would send it to `max.apt` instead.
+
+### Terms
+
+- **Domain Name**: A domain name is a string of UTF8 characters.
+- **Subdomain Name**: A domain name is a string of UTF8 characters.
+- **Name**: When we refer to a `name` we mean either `subdomain_name` or `domain_name`.
+  This is used when describing something that applies to both `Domains` _**and**_ `Subdomains`.
+- **Character Set**: In order to allow for a safe rollout of names, we are breaking UTF8
+- **Glyph**: A single character in a character set. For example, `a` is a single glyph in the `latin` character set,
+  and `你好` is two glyphs in the `simplified_chinese` character set. We specifically use `glyph` instead of `letter` to
+  help avoid confusion when dealing with character sets where the term `letter` is not applicable, such
+  as `simplified_chinese`, or `emojis`, such as this unicorn: 🦄. Aptos is a unicorn-friendly zone. Neigh!
+
+### Character Sets
+
+With great UTF8 flexibility comes great UTF8 pain. In order to avoid this pain out of the gate, we are launching with a
+limited character set, which parallels ASCII English letters and numbers with the characters it supports. We are
+referring to
+this character set as `latin`.
+We will expand both the number of character sets we support, and the glyphs within each one, as quickly as possible to
+do safely.
+
+For languages where glyphs may have an upper and lower case, all names are only allowed to be registered using lower
+case letters.
+
+We will only allow registering domains which are compatible with [RFC 5890](https://www.rfc-editor.org/rfc/rfc5890).
+
+Names are currently verified following the below rules:
+
+1. characters `a-z`, `0-9`, and `-` are allowed.
+2. Names may not begin or end with a `-`.
+3. Names may not contain more than 63 characters.
+
+### On-Chain Representation
+
+`Domains` and `Subdomains` are represented on-chain in two ways:
+
+1. As an `NFT`, following our Token Standard. When a name is registered, an NFT resource is created and `moved` into the
```

Comment:
probably good to call this a digital asset and point to aip-11 https://github.com/aptos-foundation/AIPs/edit/main/aips/aip-11.md